### DISCUSSION TITLE
.or(perseus_routes)?

### DISCUSSION THREAD

**Discussion by ghost** [framesurge/perseus#222] [Q&A]
In [examples/core/custom_server/src/main.rs](https://github.com/framesurge/perseus/blob/main/examples/core/custom_server/src/main.rs) there's "let routes = api_route.or(perseus_routes);". Why does it need .or(perseus_routes)?

**Comment by lukechu10** [comment]
The comment on top describes why.

**Comment by arctic-hen7** [comment]
That's Warp syntax, you should take a look at their docs for more information.

**Reply by ghost** [reply]
Can we use warp's path instead of perseus' [build state and build path](https://github.com/framesurge/perseus/blob/main/examples/website/state_generation/src/main.rs)?

**Reply by lukechu10** [reply]
These two are completely different concepts. Please take a look at the docs for both warp and perseus because they explain which each one is for.

Basically, warp routes are just server endpoints whereas build state and build path is for running logic at build-time to be used in your perseus app.
